main.go

```diff
@@ -46,22 +50,37 @@ var RootCmd = &cobra.Command{
 	Example: "heketi --config=/config/file/path/",
 	Run: func(cmd *cobra.Command, args []string) {
 		fmt.Printf("Heketi %v\n", HEKETI_VERSION)
-		if !showVersion {
+		if showVersion {
+			// Quit here if all we needed to do was show version
+			os.Exit(0)
+		} else {
 			// Check configuration file was given
-			if configfile == "" {
+			if configfile == "" && editMode == false {
 				fmt.Fprintln(os.Stderr, "Please provide configuration file")
 				os.Exit(1)
 			}
-		} else {
-			// Quit here if all we needed to do was show version
-			os.Exit(0)
+			// Check args for edit mode
+			if editMode == true {
```

Comment:
Instead of nested ifs perhaps a switch statement would be more idomatic Go.

Reply:
This is removed with sub command approach.